### PR TITLE
fix (posts): color overlay for featured images are now visible

### LINKED ISSUE #2351: Posts block issue - Featured image hover color not showing
**Describe the bug**
When adding a hover color (either single or gradient) for Posts block > Style > Featured Image and hovering the image in frontend, it doesn't show the hover color.

**To Reproduce**
1. Add a Posts block
2. Choose any layout
3. Make sure to add a featured image in your posts
4. In Inspector > Style > Featured Image, click Hover State and select Hover
5. Add a color for Hover
6. Click Hover State again and choose Normal
7. Hover any featured image in your Posts block
8. Observe if the hover color is appearing or not
9. Save page and check frontend
10. Repeat steps 7-8

**Expected behavior**
Hover color should appear upon hovering featured image in Posts block

**Screenshots**

https://user-images.githubusercontent.com/103395655/182751481-9611f10b-8fe9-4d4c-b63e-7beff5083483.mp4

### PR DESCRIPTION
**Pull request by prconcepcion**
Fixes #2351

### CODE REVIEW

**Comment by prconcepcion** [comment]
closes #2351

File: src/block-components/icon/style.js [hunk 1]
```diff
@@ -152,8 +167,12 @@ const Styles = props => {
...
 				} }
 				hoverSelectorCallback={ getAttribute => `${ selector }:hover #linear-gradient-${ getAttribute( 'uniqueId' ) }` }
-				dependencies={ [ 'iconColorType', 'iconColor1', 'iconColor2' ] }
-			/>
+				dependencies={ [
+					'iconColorType',
+					'iconColor1',
+					'iconColor2',
+					...dependencies,
+				] }			/>
```
Comment (bfintal) [inline src/block-components/icon/style.js:175]:
```suggestion
				] }
			/>
```

File: src/block-components/helpers/size/style.js [hunk 2]
```diff
@@ -213,7 +218,11 @@ const Styles = props => {
 				valueCallback={ value => {
 					return value.startsWith( 'auto' ) ? 'auto' : value
 				} }
-				dependencies={ [ 'horizontalAlign', 'width' ] }
+				dependencies={ [
+					'horizontalAlign',
+					 'width',
+					 ...dependencies,
+					 ] }
```
Comment (bfintal) [inline src/block-components/helpers/size/style.js:225]:
```suggestion
				dependencies={ [
					'horizontalAlign',
					 'width',
					 ...dependencies,
				] }
```

File: src/block-components/helpers/backgrounds/style.js [hunk 3]
```diff
@@ -118,7 +127,11 @@ const Styles = props => {
 					}
 					return value
 				} }
-				dependencies={ [ 'backgroundCustomSize', 'backgroundCustomSizeUnit' ] }
+				dependencies={ [
+					'backgroundCustomSize',
+					 'backgroundCustomSizeUnit',
+					 ...dependencies,
+					 ] }
```
Comment (bfintal) [inline src/block-components/helpers/backgrounds/style.js:134]:
```suggestion
				dependencies={ [
					'backgroundCustomSize',
					 'backgroundCustomSizeUnit',
					 ...dependencies,
				] }
```

**Comment by andeng1106** [comment]
@prconcepcion 
Will move this to backlog.

Issues (affected/started happening after fix):

- [ ] (started in fix) When hovering in editor, the overlay color overspills outside the featured image (editor issue)

https://user-images.githubusercontent.com/103395655/206955724-a3c5b2db-96cf-4e42-9d67-2d36464fe295.mp4

- [ ]  (affected) Hover state settings is not showing if "Hover State" display/mode is selected (editor issue)
might be related to this issue (fixed by Vivo, **please coordinate**): https://www.notion.so/Hover-and-Parent-Hovered-state-display-doesn-t-seem-to-be-showing-in-editor-anymore-see-notes-for-m-20c3d6ee901d415183a3eabfc13b5e55

- [ ]  (affected) Some Featured Image panel settings with hover state are no longer working as expected (both frontend/editor issue)

https://user-images.githubusercontent.com/103395655/206956254-b4595b15-bb7f-4b7f-84a0-a88aeb9f71af.mp4

- Question: Is it really necessary for the user to save page again for the fix to appear?

**Comment by bfintal** [comment]
@prconcepcion 2 remaining issues:

1. When hovering in editor, the overlay color overspills outside the featured image (editor only issue)

   https://user-images.githubusercontent.com/103395655/206955724-a3c5b2db-96cf-4e42-9d67-2d36464fe295.mp4

2. Some Featured Image panel settings with hover state are no longer working as expected (both frontend/editor issue)

   https://user-images.githubusercontent.com/103395655/206956254-b4595b15-bb7f-4b7f-84a0-a88aeb9f71af.mp4

**Comment by andeng1106** [comment]
Remaining issue:
- Gradient colors not displaying when hovering image (issue both in frontend and editor)
   - Gradient settings might also be affected (Gradient Overlay Settings, Overlay Opacity)
   

https://user-images.githubusercontent.com/103395655/211692026-6daba923-df8d-4abf-8c6e-c0fe24a6d709.mp4

**Comment by andeng1106** [comment]
Remaining issue:

1. It's causing a block error when adding a posts block w/ featured image set to gradient colors:
   - Tried adding from local site with v3.6.3 build and pasted on local site w/ PR 3533 build
   - Tried loading old post block created using from PR 2446 (performance build) then change build to PR 3533 then encounters a block error
2. Still not getting the expected "Hover State" gradient overlay colors on editor (see video)
   

https://user-images.githubusercontent.com/103395655/211945126-85aaa1de-84c4-48da-a840-970364337505.mp4

**Comment by andeng1106** [comment]
Block error doesn't happen anymore but there's still a remaining issue:
- Gradient hover overlay color is not showing up when hovering Featured image (issue both in frontend and editor)
	- Tried it with a new and old existing block and it doesn't work on both frontend and editor :((

https://user-images.githubusercontent.com/103395655/212285907-fac8b9e1-c436-4870-b3de-40d90de0e704.mp4

**Pull request merged**